src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCache.scala

```diff
@@ -41,91 +50,113 @@ class GoogleKeyCache(
         this
       ))
 
-    googleStorageDAO.createBucket(googleServicesConfig.serviceAccountClientProject, googleServicesConfig.googleKeyCacheConfig.bucketName).recover {
-      case t: GoogleJsonResponseException if t.getDetails.getMessage.contains("You already own this bucket") && t.getDetails.getCode == 409 => ()
-    } flatMap { _ =>
-      googleStorageDAO.setBucketLifecycle(googleServicesConfig.googleKeyCacheConfig.bucketName, googleServicesConfig.googleKeyCacheConfig.retiredKeyMaxAge)
+    googleStorageAlg
+      .createBucket(googleServicesConfig.serviceAccountClientProject, googleServicesConfig.googleKeyCacheConfig.bucketName, List.empty)
+      .recoverWith {
+        case t: StorageException if t.getCode == 409 && t.getMessage.contains("You already own this bucket") =>
+          IO(logger.info(t.getMessage))
+      } flatMap { _ =>
+      val lifecycleCondition = BucketInfo.LifecycleRule.LifecycleCondition
+        .newBuilder()
+        .setAge(googleServicesConfig.googleKeyCacheConfig.retiredKeyMaxAge)
+        .build()
+      val lifecycleRule = new LifecycleRule(LifecycleRule.LifecycleAction.newDeleteAction(), lifecycleCondition)
+      googleStorageAlg.setBucketLifecycle(googleServicesConfig.googleKeyCacheConfig.bucketName, List(lifecycleRule))
     }
   }
 
-  override def getKey(pet: PetServiceAccount): Future[String] = {
-    val retrievedKeys = for {
-      keyObjects <- googleStorageDAO.listObjectsWithPrefix(
-        googleServicesConfig.googleKeyCacheConfig.bucketName,
-        keyNamePrefix(pet.id.project, pet.serviceAccount.email))
-      keys <- googleIamDAO.listUserManagedServiceAccountKeys(pet.id.project, pet.serviceAccount.email).map(_.toList)
-      cleanedKeyObjects <- cleanupUnknownKeys(pet, keyObjects, keys)
-    } yield (cleanedKeyObjects, keys)
-
-    retrievedKeys.flatMap {
-      case (Nil, _) => furnishNewKey(pet) //mismatch. there were no keys found in the bucket, but there may be keys on the service account
-      case (_, Nil) => furnishNewKey(pet) //mismatch. there were no keys found on the service account, but there may be keys in the bucket
-      case (keyObjects, serviceAccountKeys) => retrieveActiveKey(pet, keyObjects, serviceAccountKeys)
+  override def getKey(pet: PetServiceAccount): IO[String] = {
```

Comment:
I found some of the logic in here hard to follow so I took a stab at it myself. Let me know what you think or just accept the PR if you like it: https://github.com/broadinstitute/sam/pull/271